builtin/logical/pki/path_roles.go

```diff
@@ -339,7 +339,7 @@ func (b *backend) pathRoleCreate(
 		// If they are using the system default, cap it to the role max;
 		// if it was specified on the command line, make it an error
 		if len(entry.TTL) == 0 {
-			ttl = maxTTL
+			entry.TTL = entry.MaxTTL
```

Comment:
On line 318, `maxTTL` is set but notice that `entry.MaxTTL` is not set. So at line 342, if we do `entry.TTL = entry.MaxTTL`, the RHS may be 0.

You are right about setting the values in `entry` before storing it.
By the time we reach line 349, both `ttl` and `maxTTL` will have the desired values to be set into the `entry` object. So if we just do a `entry.TTL = ttl; entry.MaxTTL = maxTTL` there, it would suffice. I _think_ this might have been the original intention.